### PR TITLE
Feature/issue 1978 remote checkpoint interference

### PR DESCRIPTION
**Pull request by tleyden**
Fixes https://github.com/couchbase/sync_gateway/issues/1978

### CODE REVIEW

**Comment by tleyden** [comment]
Note: make sure http://uberjenkins.sc.couchbase.com/view/QE%20Dev/job/syncgateway-functional-tests-dev/134/console passes before merging

**Comment by tleyden** [comment]
N/m, looks like the jenkins functional test passed

File: synctube.go [hunk 1]
```diff
@@ -140,13 +140,53 @@ func (r *Replication) shutdownEventChannel() {
...
+
+	// `Write` expects bytes. If you have a string `s`,
+	// use `[]byte(s)` to coerce it to bytes.
+	h.Write([]byte(replicationParamsJson))
+
+	// This gets the finalized hash result as a byte
+	// slice. The argument to `Sum` can be used to append
+	// to an existing byte slice: it usually isn't needed.
+	bs := h.Sum(nil)
+
```
Comment (adamcfraser) [inline synctube.go]: For this use case I thought we could just use sha1.Sum([]byte(replicationParamsJson), and not need the New/Write/Sum.  e.g. https://golang.org/pkg/crypto/sha1/#Sum

**Comment by tleyden** [comment]
@adamcfraser should be ready to merge now

**Pull request merged**